### PR TITLE
Test Cleanup - Refactor out uses of deprecated `callAPIAndDocument()`

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Stop using a function that no longer does what it says it does.

### CODE REVIEW

File: tests/phpunit/api/v3/EventTest.php [hunk 1]
```diff
@@ -463,11 +459,6 @@ public function testEventGetCustomContactRefFieldCRM16036() {
     ]);
     $contact_id = $contact_save_result['id'];
 
-    // I have no clue what this $subfile is about. I just copied it from another
-    // unit test.
```
Comment (colemanw) [inline tests/phpunit/api/v3/EventTest.php:467]: Gotta love the honesty!

File: tests/phpunit/api/v3/ContactTest.php [hunk 2]
```diff
@@ -686,7 +686,7 @@ public function testCreateWithCustom(int $version): void {
     $params = $this->_params;
     $params['custom_' . $ids['custom_field_id']] = 'custom string';
     $description = 'This demonstrates setting a custom field through the API.';
-    $result = $this->callAPIAndDocument($this->_entity, 'create', $params, __FUNCTION__, __FILE__, $description);
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ContactTest.php:689]: Looks like they forgot to copy $subfile here. Clearly an oversight!

(i.e. could remove the $description var in line 688 - I assume your regex/script is also looking for subfile so doesn't see it)

File: tests/phpunit/api/v3/ContactTest.php [hunk 3]
```diff
@@ -3120,8 +3093,7 @@ public function testContactGetFormatCountOnly(int $version): void {
     $description = "This demonstrates use of the 'getCount' action.
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ContactTest.php:3093]: Ditto

File: tests/phpunit/api/v3/ContactTest.php [hunk 4]
```diff
@@ -4028,7 +3992,7 @@ public function testGetlistInvalidID(): void {
    */
   public function testGetActions(): void {
     $description = 'Getting the available actions for an entity.';
```
Comment (demeritcowboy) [inline tests/phpunit/api/v3/ContactTest.php:3994]: ditto3

Comment (demeritcowboy) [reply tests/phpunit/api/v3/ContactTest.php:3994]: I do see a couple more but have stopped looking. It's all tests so NFC.

**Comment by colemanw** [comment]
Thanks for your descriptive comments @demeritcowboy! Updated.

**Comment by totten** [comment]
# 🎉

**Pull request merged**